### PR TITLE
[mlir][spirv] Fix spirv dialect to support Specialization constants as GlobalVar initializer

### PR DESCRIPTION
**Pull request by drprajap**
Changes include:

- spirv serialization and deserialization needs handling in cases when GlobalVariableOp initializer is defined using spirv  SpecConstant or SpecConstantComposite op, currently even though it allows SpecConst, it only looked up in for GlobalVariable Map to find initializer symbol reference, change is fixing this and extending the support to SpecConstantComposite as an initializer.
- Adds tests to make sure GlobalVariable can be initialized using specialized constants.

### CODE REVIEW

**Comment by drprajap** [comment]
@mshahneo , @silee2 - FYI

File: mlir/lib/Target/SPIRV/Deserialization/Deserializer.cpp
Comment (kuhar) [inline mlir/lib/Target/SPIRV/Deserialization/Deserializer.cpp:652]: Since this is accessing cast results unconditionally, I'd expect to see `cast<T>(op)` used instead of `dynamic_cast<T>`

**Review by kuhar — APPROVED** [review]
LGTM but please wait for @antiagainst to take a look to

**Comment by drprajap** [comment]
> LGTM but please wait for @antiagainst to take a look to

Thank you :), sure I will wait for 2nd approval.

**Review by antiagainst — APPROVED** [review]

**Comment by antiagainst** [comment]
Thanks for the contribution! Sorry about the delay in reviews. I do have some style-ish comments but given this has been a while I just went ahead fixed them and will land this path directly. Thanks again!

**Comment by drprajap** [comment]
> Thanks for the contribution! Sorry about the delay in reviews. I do have some style-ish comments but given this has been a while I just went ahead fixed them and will land this path directly. Thanks again!

Thanks for the review and merging it. No worries about the delay, it was holidays time, so understood..

**Pull request merged**
